### PR TITLE
Deprecate pgr analyze one way

### LINKED ISSUE #2754: Deprecate pgr_analyzeOneWay in 3.8
Created: v2.0.0
Reason for deprecation:
- Returns  "OK"/"FAIL" 
- A notice is given with some statistics (instead of columns with the statistics)
- Statistics are just a count of ...

pgRouting has grown and has more graph analysis like ``pgr_degree``.

### PR DESCRIPTION
**Pull request by cvvergara**
Fixes #2754  

Changes proposed in this pull request:
- Removed use on documentation
- Removed use on pgtap
- Removed references to documentation file
- Added migration instructions

@pgRouting/admins

## Summary by CodeRabbit

- **Documentation**
  - Updated release notes, migration guides, and API documentation to indicate that the `pgr_analyzeOneWay` function is deprecated as of v3.8.0. Guidance on adjusting to these changes is now included.
- **New Features**
  - Introduced new SQL queries that enhance the analysis of one-way edge conditions based on cost criteria.
- **Tests**
  - Revised test cases to remove obsolete one-way analysis checks and update expected outcomes.

### CODE REVIEW

**Review by robe2 — APPROVED** [review]

File: docqueries/src/migration.pg [hunk 1]
```diff
@@ -401,3 +401,21 @@ FROM results JOIN deadends d USING (geom);
...
+/* --OneWay2 */
+SELECT id, cost < 0 OR reverse_cost<0 AS is_OneWway
+FROM pgr_bridges('SELECT id, source, target, cost, reverse_cost FROM edges')
+JOIN edges ON (edge = id);
+/* --OneWay3 */
+SELECT id, cost, reverse_cost
+FROM pgr_bridges('SELECT id, source, target, cost, reverse_cost FROM edges')
+JOIN edges ON (edge = id)
+WHERE cost < 0 OR reverse_cost<0;
+/* --OneWay4 */
```
[the remark this answers is not in the document]
Comment (cvvergara) [reply docqueries/src/migration.pg:421]: The last one is the one that separates another migration
